internal/services/containers/migration/kubernetes_cluster.go

```diff
@@ -1223,3 +1222,1312 @@ func (k KubernetesClusterV0ToV1) Schema() map[string]*pluginsdk.Schema {
 		},
 	}
 }
+
+func (k KubernetesClusterV1ToV2) UpgradeFunc() pluginsdk.StateUpgraderFunc {
+	// since `server_app_secret` isn't returned from the AKS API we need to populate that value in the new `azure_active_directory_role_based_access_control` block
+	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+		secretRaw := ""
+		if rbac, ok := rawState["role_based_access_control"]; ok {
+			rbacRaw := rbac.([]interface{})[0].(map[string]interface{})
+			if aad, ok := rbacRaw["azure_active_directory"]; ok {
+				aadRaw := aad.([]interface{})
+				if len(aadRaw) == 0 {
+					return rawState, nil
+				}
+				aadMap := aadRaw[0].(map[string]interface{})
+				if secret, ok := aadMap["server_app_secret"]; ok {
+					log.Printf("[DEBUG] found value for `role_based_access_control.0.azure_active_directory.0.server_app_secret`")
+					secretRaw = secret.(string)
+				}
+			}
+		}
+
+		if secretRaw != "" {
+			log.Printf("[DEBUG] copying value to `azure_active_directory_role_based_access_control.0.server_app_secret`")
+			rawState["azure_active_directory_role_based_access_control"] = []interface{}{
+				0: map[string]interface{}{
```

Comment:
this doesn't need indexing since it's defining a new slice fwiw:

```suggestion
				map[string]interface{}{
```